### PR TITLE
Feature/auth paypal

### PR DESCRIPTION
**Pull request by denisnikulin**
login check,
login tracking,
events logging in mongo
TODO: automated token handling.

### CODE REVIEW

File: src/env.js [hunk 1]
```diff
@@ -24,4 +24,25 @@ module.exports = envalid.cleanEnv(process.env, {
     default: 4001,
     desc: 'subscription port (websocket)',
   }),
+  PAYPAL_MODE: envalid.str({
+    default: 'sandbox',
+    desc: 'paypal environment sandbox | live',
+  }),
+  PAYPAL_CLIENT_ID: envalid.str({
+    default: '',
```
Comment (kachkaev) [inline src/env.js]: Should `PAYPAL_CLIENT_ID` and `PAYPAL_CLIENT_SECRET` have a default value? Or should `npm start` fail with an error from envalid when not defined?

File: src/env.js [hunk 2]
```diff
@@ -24,4 +24,25 @@ module.exports = envalid.cleanEnv(process.env, {
...
+  PAYPAL_CLIENT_SECRET: envalid.str({
+    default: '',
+    desc: 'paypal client secret',
+  }),
+  PAYPAL_REDIRECT_URL: envalid.url({
+    default: '/login/callback',
+    desc: 'paypal redirect url',
+  }),
+  MONGO_URL: envalid.str({
+    default: 'mongodb://localhost/them',
```
Comment (kachkaev) [inline src/env.js]: `mongodb://localhost/them` → `mongodb://localhost/callthemonline` ?

File: src/resolvers.js [hunk 3]
```diff
@@ -11,13 +12,33 @@ const resolvers = {
     unixTimestamp() {
       return Math.round(+new Date() / 1000);
     },
+
+    user(obj, args, { token }) {
+      if (!token) { return null; }
+      return {
+        id: 111111,
```
Comment (kachkaev) [inline src/resolvers.js]: OK for MVP? Or is this pre-MVP?

**Review by kachkaev — CHANGES_REQUESTED** [review]

File: src/resolvers.js (hunk 3 above)
Comment (denisnikulin) [reply src/resolvers.js]: I will fix it by implementing token on both sides. and only then we should considr merging.

File: src/env.js [hunk 4]
```diff
@@ -24,4 +24,25 @@ module.exports = envalid.cleanEnv(process.env, {
     default: 4001,
     desc: 'subscription port (websocket)',
   }),
+  PAYPAL_MODE: envalid.str({
+    default: 'sandbox',
+    desc: 'paypal environment sandbox | live',
+  }),
+  PAYPAL_CLIENT_ID: envalid.str({
+    default: 'NULL',
```
Comment (kachkaev) [inline src/env.js:32]: You can just remove this line and the same one below. NULL as string is a valid number otherwise.

**Comment by kachkaev** [comment]
index

**Review by kachkaev — APPROVED** [review]

File: src/env.js (hunk 4 above)
Comment (denisnikulin) [reply src/env.js:32]: I made this on purpose.

**Comment by denisnikulin** [comment]
sorry the branch `auth` was created to test CI if `/` can be a part of docker container in repo.

**Review by kachkaev — APPROVED** [review]

**Pull request merged**